chainstate/tx-verifier/src/transaction_verifier/input_output_policy/constraints_accumulator.rs

```diff
@@ -13,705 +13,584 @@
-    // Check it's an error.
-    // Then create and check valid case.
-    #[rstest]
-    #[trace]
-    #[case(Seed::from_entropy())]
-    fn try_to_unlock_coins_with_smaller_timelock(#[case] seed: Seed) {
-        let mut rng = make_seedable_rng(seed);
-
-        let chain_config = common::chain::config::Builder::new(ChainType::Mainnet)
-            .consensus_upgrades(NetUpgrades::regtest_with_pos())
-            .build();
-        let required_maturity_distance =
-            chain_config.staking_pool_spend_maturity_distance(BlockHeight::new(1));
-
-        let pool_id = PoolId::new(H256::zero());
-        let staked_atoms = rng.gen_range(100..1000);
-        let less_than_staked_amount = Amount::from_atoms(rng.gen_range(1..staked_atoms));
-        let stake_pool_data = create_stake_pool_data(&mut rng, staked_atoms);
-
-        let pledge_getter = |_| Ok(Some(Amount::from_atoms(staked_atoms)));
-
-        let inputs = vec![
-            TxInput::from_utxo(
-                OutPointSourceId::BlockReward(Id::new(H256::random_using(&mut rng))),
-                0,
-            ),
-            TxInput::from_utxo(
-                OutPointSourceId::BlockReward(Id::new(H256::random_using(&mut rng))),
-                1,
-            ),
-        ];
-        let inputs_utxos = vec![
-            Some(TxOutput::CreateStakePool(
-                pool_id,
-                Box::new(stake_pool_data),
-            )),
-            Some(TxOutput::Transfer(
-                OutputValue::Coin(less_than_staked_amount),
-                Destination::AnyoneCanSpend,
-            )),
-        ];
-
-        let outputs = vec![
-            TxOutput::LockThenTransfer(
-                OutputValue::Coin(Amount::from_atoms(staked_atoms - 10)),
-                Destination::AnyoneCanSpend,
-                OutputTimeLock::ForBlockCount(required_maturity_distance.to_int() as u64),
-            ),
-            TxOutput::LockThenTransfer(
-                OutputValue::Coin(Amount::from_atoms(10)),
-                Destination::AnyoneCanSpend,
-                OutputTimeLock::ForBlockCount(required_maturity_distance.to_int() as u64 - 1),
-            ),
-            TxOutput::Transfer(
-                OutputValue::Coin(less_than_staked_amount),
-                Destination::AnyoneCanSpend,
-            ),
-        ];
-
-        let mut constraints_accumulator = ConstrainedValueAccumulator::new();
-
-        constraints_accumulator
-            .process_inputs(
-                &chain_config,
-                BlockHeight::new(1),
-                pledge_getter,
-                &inputs,
-                &inputs_utxos,
-            )
-            .unwrap();
-
-        let result = constraints_accumulator.process_outputs(&chain_config, &outputs).unwrap_err();
-        assert_eq!(
-            result,
-            IOPolicyError::AttemptToPrintMoneyOrViolateTimelockConstraints
-        );
+    /// Return accumulated coins that are left
+    pub fn consume(
```

Comment:
I would call this "map_into_block_fees", and write the documentation "Map/project the accumulator into an object that represents the accumulated block fees. We call this projection because the result of this projection is a subgroup of the `ConstrainedValueAccumulator` group under the 'accumulation' operation, where we can accumulate fees as well in the subgroup."

I'm OK with finding a better way to phrase this.

Currently it's OK to keep using the Fee type alias. Though in general, it'd be more meaningful to use a dedicated struct for this and implement `From<AccumulatedBlockFees> for ConstrainedValueAccumulator`. Maybe later.